examples/zkapps/05-common-types-and-functions/src/main.ts

```diff
@@ -26,301 +25,295 @@ import {
+const char2 = Character.fromString('d');
 
-  // --------------------------------------
+console.log(`char1: ${char1.toString()}`);
+console.log(`char1 == char2: ${char1.equals(char2).toString()}`);
+console.log(`Fields in char1: ${char1.toFields().length}`);
+console.log("// --------------------------------------");
 
-  const str1 = CircuitString.fromString('abc..xyz');
-  console.log(`str1: ${str1.toString()}`);
-  console.log(`Fields in str1: ${str1.toFields().length}`);
+// --------------------------------------
 
-  // --------------------------------------
+const str1 = CircuitString.fromString('abc..xyz');
+console.log(`str1: ${str1.toString()}`);
+console.log(`Fields in str1: ${str1.toFields().length}`);
 
-  const privateKey = PrivateKey.random();
-  const publicKey = privateKey.toPublicKey();
+// --------------------------------------
 
-  const data1 = char2.toFields().concat(signedNumSum.toFields())
-  const data2 = char1.toFields().concat(str1.toFields());
+const zkAppPrivateKey = PrivateKey.random();
+const zkAppPublicKey = zkAppPrivateKey.toPublicKey();
 
-  const signature = Signature.create(privateKey, data2);
+const data1 = char2.toFields().concat(signedNumSum.toFields())
+const data2 = char1.toFields().concat(str1.toFields());
 
-  const verifiedData1 = signature.verify(publicKey, data1);
-  const verifiedData2 = signature.verify(publicKey, data2);
+const signature = Signature.create(zkAppPrivateKey, data2);
 
-  console.log(`private key: ${privateKey.toBase58()}`);
-  console.log(`public key: ${publicKey.toBase58()}`);
-  console.log(`Fields in private key: ${privateKey.toFields().length}`);
-  console.log(`Fields in public key: ${publicKey.toFields().length}`);
+const verifiedData1 = signature.verify(zkAppPublicKey, data1);
+const verifiedData2 = signature.verify(zkAppPublicKey, data2);
 
-  console.log(`signature verified for data1: ${verifiedData1.toString()}`);
-  console.log(`signature verified for data2: ${verifiedData2.toString()}`);
+console.log(`private key: ${zkAppPrivateKey.toBase58()}`);
+console.log(`public key: ${zkAppPublicKey.toBase58()}`);
+console.log(`Fields in private key: ${zkAppPrivateKey.toFields().length}`);
+console.log(`Fields in public key: ${zkAppPublicKey.toFields().length}`);
 
-  console.log(`Fields in signature: ${signature.toFields().length}`);
-  console.log("// --------------------------------------");
+console.log(`signature verified for data1: ${verifiedData1.toString()}`);
+console.log(`signature verified for data2: ${verifiedData2.toString()}`);
 
-  // --------------------------------------
+console.log(`Fields in signature: ${signature.toFields().length}`);
+console.log("// --------------------------------------");
 
-  class Point extends CircuitValue {
-    @prop x: Field;
-    @prop y: Field;
+// --------------------------------------
 
-    static fromField(x: Field, y: Field) {
-      return new Point(x, y);
-    }
+class Point extends CircuitValue {
```

Comment:
it would be nice if this used `Struct` instead of (deprecated) `CircuitValue`, but we can also save that for later

Reply:
Agreed! I'd need a quick intro or referece to Struct but yes let's do that another time